arch/arm/soc/nxp_lpc/lpc54xxx/soc.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for nxp_lpc54114 platform
+ *
+ * This module provides routines to initialize and support board-level
+ * hardware for the nxp_lpc54114 platform.
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <uart.h>
+#include <linker/sections.h>
+#include <arch/cpu.h>
+#include <cortex_m/exc.h>
+#include <clock_config.h>
+#include <fsl_lpc_power.h>
+#include <fsl_clock.h>
+#include <fsl_common.h>
+#include <fsl_device_registers.h>
+
+/**
+ *
+ * @brief Initialize the system clock
+ *
+ * @return N/A
+ *
+ */
+
+static ALWAYS_INLINE void clkInit(void)
+{
+    /* Set up the clock sources */
+
+	/* Ensure FRO is on */
+	POWER_DisablePD(kPDRUNCFG_PD_FRO_EN);
+
+    /*
+     * Switch to FRO 12MHz first to ensure we can change voltage without
+     * accidentally being below the voltage for current speed.
+     */
+	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
+
+    /* Set FLASH wait states for core */
+	CLOCK_SetFLASHAccessCyclesForFreq(BOARD_BOOTCLOCKFROHF48M_CORE_CLOCK);
+
+    /* Set up high frequency FRO output to selected frequency */
+	CLOCK_SetupFROClocking(BOARD_BOOTCLOCKFROHF48M_CORE_CLOCK);
+
+    /* Set up dividers */
+    /* Set AHBCLKDIV divider to value 1 */
+	CLOCK_SetClkDiv(kCLOCK_DivAhbClk, 1U, false);
+
+    /* Set up clock selectors - Attach clocks to the peripheries */
+    /* Switch MAIN_CLK to FRO_HF */
+	CLOCK_AttachClk(kFRO_HF_to_MAIN_CLK);
+
+    /* Attach 12 MHz clock to FLEXCOMM0 */
+	CLOCK_AttachClk(kFRO12M_to_FLEXCOMM0);
+
+    /* Enable clock for gpio*/
+	CLOCK_EnableClock(kCLOCK_Gpio0);
```

Comment:
#ifdef CONFIG_GPIO_LPC_MCUX_PORT0 for enabling GPIO0 clock.
    CLOCK_EnableClock(kCLOCK_Gpio0);
#endif

same for Gpio1?